### ISSUE TITLE
Custom title bar on Windows/Linux

### ISSUE THREAD

**Issue by erroid** [atom/atom#13249]
Using dark theme and having white title bar is not nice. Please make it more consistent with rest of the app like in Adobe PS or  Brackets (also look #4599).

**Comment by 50Wliu** [comment]
Blocked on electron/electron#765.

**Comment by erroid** [comment]
How about Windows?

**Comment by 50Wliu** [comment]
Windows does not have a builtin dark theme, and no other program that I can find updates their title bar based on the accent color.

If you mean high contrast, there are a bunch of different Electron issues related to that as well.

**Comment by erroid** [comment]
I talk about title bar (image attached). There is a lot of programs that use their own designed title bars instead of those delivered by windows system, here are some examples:

![atom-title](https://cloud.githubusercontent.com/assets/10036579/20399068/e68ad55c-acef-11e6-8369-6dd8a8d71f52.png)

**Comment by jerone** [comment]
A few more on Windows programs that have a themed title-bar:
* Visual Studio.
* All Microsoft Office programs.

**Comment by alexlur** [comment]
Related: This feature has just [landed](https://github.com/Microsoft/vscode/issues/15098) in Visual Studio Code Insiders.

**Comment by 50Wliu** [comment]
Cool, looks like I misunderstood the issue to be talking about Window's accent color vs the Atom theme.  I completely forgot about MS Office.  Going to cc @simurai then to see what can be done.

**Comment by simurai** [comment]
Electron has [an option](http://electron.atom.io/docs/api/frameless-window/#alternatives-on-macos) for macOS to hide the title bar but keep the window controls (traffic lights). Not sure if that will also be available for Windows/Linux at some point.

I guess currently the only way would be to go completely frameless and then re-create the title bar with the window controls that can be themed. Maybe ok for Windows. For Linux with all the different flavors it would probably be a bigger effort.

**Comment by tj** [comment]
It would be awesome to optionally completely remove them, I think most people know how to close a window and minimize etc via shortcuts.

**Comment by Yajo** [comment]
On 🐧 this can help a lot and it seems very simple: https://eischmann.wordpress.com/2017/01/10/dark-title-bars-for-apps-with-dark-ui/

Quoting:

> They just need to set an x window property GTK_THEME_VARIANT=dark

**Comment by JLCCI** [comment]
+1

**Comment by xtoolkit** [comment]
hi :)
i customize atom!
![screenshot 41](https://cloud.githubusercontent.com/assets/8210666/23853363/7ca98698-0801-11e7-9410-1863fc1f8752.png)
demo :[video](http://arshen.ir/bandicam11.mp4)

**Comment by kawaiidon** [comment]
@xtoolkit  how??

**Comment by xtoolkit** [comment]
@kawaiidon
my project pull https://github.com/xtoolkit/arshen-ac/pull/1 . this project request https://github.com/atom/atom/pull/14003 pull!

**Comment by Hammster** [comment]
@xtoolkit really hope this get merged ASAP, I mean all my electron apps have custom title bars and sudden atom does not, thank you for the work :+1:

**Comment by lethal-bacon** [comment]
How were you able to do this?

**Comment by Am3ra** [comment]
@simurai Any update on removing all the chrome?

**Comment by simurai** [comment]
> @simurai Any update on removing all the chrome?

I don't know enough to take this on myself. 😇  If somebody creates a PR, it might can be considered. Just hiding the menu (instead of a custom title bar) probably has a better chance since it's less complex.

**Comment by sindrets** [comment]
@erroid, @simurai, @Am3ra, @Hammster I created a [package ](https://atom.io/packages/title-bar-replacer)for Windows that adds a custom, theme-aware, functional title bar and menu bar to Atom. It also has functionality to remove the native title bar. Currently the removal of the native title bar only works on Windows, but I assume it wouldn't be too hard to get it working on Linux. Check it out if you're interested.

![title bar replacer](https://user-images.githubusercontent.com/2786478/28589179-4dd010ae-717d-11e7-82ab-6a6d14ab4c25.PNG)

**Comment by kawaiidon** [comment]
@sindrets epic! thenks

**Comment by Xenpai** [comment]
@sindrets I'm curious about the themes you're using in the preview gif

**Comment by sindrets** [comment]
@Xenpai the dark UI theme is One Dark with [Nord Atom Syntax](https://atom.io/themes/nord-atom-syntax). The light UI theme is One Light with [Br4ckets Light](https://atom.io/themes/br4ckets-light) syntax.

**Comment by Xenpai** [comment]
@sindrets Thanks a lot!

**Comment by silverkorn** [comment]
@sindrets, works really great and it's pretty slick on my end using Windows 7.
Little notice: Would be great to suggests/have few different icon sets, the maximize button seems slighly too big, but that's just a personal opinion.

This should be added in the official release 😄

**Comment by erroid** [comment]
@sindrets nice job! if you add kayboard navigation for menu bar it will be perfect! : ))

**Comment by jsejcksn** [comment]
Just want to add some screenshots for each of the title bar options in the macOS version of Atom for reference (using One Dark for [UI](https://github.com/atom/one-dark-ui) & [Syntax](https://github.com/atom/one-dark-syntax) Themes):

![atom-settings-title_bar](https://user-images.githubusercontent.com/868251/29735797-5437627e-89c1-11e7-95ba-ee80122bdcac.gif)
Animated gif of each of the following options (added sequence dots for reference)

![atom-settings-title_bar-native](https://user-images.githubusercontent.com/868251/29735456-b912fda0-89be-11e7-848d-6f80a968c03c.png)
native

![atom-settings-title_bar-custom](https://user-images.githubusercontent.com/868251/29735463-c015c70e-89be-11e7-96b1-2b03ac8d2b3f.png)
custom

![atom-settings-title_bar-custom_inset](https://user-images.githubusercontent.com/868251/29735464-c23b59ea-89be-11e7-8238-c3d21d1bd600.png)
custom-inset

![atom-settings-title_bar-hidden](https://user-images.githubusercontent.com/868251/29735465-c6220482-89be-11e7-8ac7-44f7569aca16.png)
hidden

**Comment by skylize** [comment]
@sindrets  
> I created a package for Windows that adds a custom, theme-aware, functional title bar and menu bar to Atom. It also has functionality to remove the native title bar. 

This package is useless for Windows 10. It adds your styled title bar, plus a bunch of padding, and then the system title bar on top of it. 😭

**Comment by sindrets** [comment]
@skylize read the entire readme on the package page. My package adds a utility for easy removal of the native title bar. I am using Windows 10 myself

**Comment by skylize** [comment]
@sindrets Oh. I see. Thanks. Can this be styled? It's more attractive than the system bar, but it's still a big huge titlebar. I would just as soon be rid of the titlebar completely.

**Comment by skylize** [comment]
The need to run the removal tool should be MUCH more prominent in the settings.

**Comment by sindrets** [comment]
@skylize for custom styling, see [this issue](https://github.com/sindrets/atom-title-bar-replacer/issues/3). I'll probably add a toggle for the title bar at some point. There's already a toggle for the menu bar.

**Comment by EugenSeverins** [comment]
Nice work @sindrets. One problem i have is dat the modifying 'atom-window.js' hangs for a couple of hours.

**Comment by sindrets** [comment]
@Severe1 It's a problem with Atom 1.21. I'm working on a patch.

**Comment by Severe1** [comment]
Ah good to see, looking forward to it, i also opened an issue on the repo, which you can use ;)

**Comment by fouram** [comment]
If this is something that is possible to do with an extension, why can't it be merged into the main code as an option? Atom looks unprofessional when the title bar doesn't match the theme IMHO. Is it possible to create a PR of this instead of having to patch every time Atom updates?

**Comment by skylize** [comment]
I think it makes reasonable sense to have most of the work for this in external packages, but we need the right hooks inside core to make it possible.

It would be great if @sindrets would make a PR for the patch to `app.asar`.

If accepted, that would reduce maintenance requirements `title-bar-replacer`, make installation of and updates to `title-bar-replacer` much more pleasant. and make it _much_ easier for somebody else to build a similar package if they wanted to.

**Comment by Menci** [comment]
Here is my titlebar implement: https://github.com/Menci/electron-titlebar

**Comment by grig0ry** [comment]
After updating Electron to version 2.0 in Atom 1.28.0, this problem will be solved? 🤔

**Comment by Hammster** [comment]
@grig0ry the electron version is not the issue, it is the lack of prioritisation. `title-bar-replacer` is the perfect example that it is easily done.

**Comment by angela-d** [comment]
Plugins aren't needed to achieve this on Linux, Atom takes on the desktop environment's theme color.

For example:
![atom](https://user-images.githubusercontent.com/33108490/40699591-cbfc05f4-639a-11e8-9581-a1788646b98a.png)

This is using the GTK theme Qogir-dark: https://www.gnome-look.org/p/1230631/

The only issue I ran into, is if the theme resides exclusively under `~/.local`, it has mixed interpretations of a complete theme. (The Atom menu bar in this case, does not get themed)

In order for all elements to be properly themed, the following conditions must be met:
- The theme itself must cover window elements, not all themes are equal
- Window elements are only darkened if the theme is under `/usr/share/themes/` - the `~/.local` path can then be used for customizations without over writing the parent theme.

This is applicable under Gnome environments with GTK themes, it may differ with other DE's.

**Comment by qubitz** [comment]
This feature would be major improvement to the look of Atom. I really hope this is on someone's list of todo.

**Comment by Hammershaft** [comment]
After a new update the title-bar package @sindrets has made needs to again be patched and then applied by users. Is there any consideration or work being done towards this as a default, an option, or at least an option for a frameless window? I think it hurts the new user experience when they see this compared to how it is handled in Visual Studio.

**Comment by Jnss98b** [comment]
Wow, after more than 3 years this simple issue still hasn't been fully resolved

**Comment by jim4067** [comment]
> hi :)
> i customize atom!
> ![screenshot 41](https://cloud.githubusercontent.com/assets/8210666/23853363/7ca98698-0801-11e7-9410-1863fc1f8752.png)
> demo :[video](http://arshen.ir/bandicam11.mp4)

how did you do it. cannot find the video

**Comment by Sporking** [comment]
I recently upgraded to Windows 10, and moved my system to dark mode. Everything worked fine except for Notepad, which blasts my eyes with blinding light every time I open it on an otherwise-dark desktop. I therefore started looking for a Notepad alternative that would work in dark mode. (I was using Metapad before I switched to Windows 10, but that is no longer maintained and also doesn't support a complete dark mode.)

The open-source Notepad++ seemed like an obvious choice, but I swiftly discovered that despite dark themes for inside editor windows, it does not have a way to make the application borders dark. (using dark themes help, but that just means that the title bar, etc. are blinding as compared to a dark-themed editor window and to the rest of my Windows desktop.) So I uninstalled Notepad++.

My next attempt is Atom. It seemed promising at first in that at least the tabs and the topmost bar (at least when Atom is the currently selected application) are dark, and it's possible to set a dark theme for the UI as opposed to just editor windows contents. However, I am surprised to discover that there seems to be no straightforward (at least to a new user like me) way to make the menu bar dark, or any of the menus it produces. I was also surprised to discover that this issue has been logged as a feature request for 3 years without any apparent progress on it, despite what appears to be some workarounds submitted by users. (I don't actually know enough about the product yet to be able to attempt workarounds like these... but then I shouldn't need to. I believe that this is basic functionality that should work out of the box without me having to become an expert in Atom internals. I don't have time to become an expert in every tool I use, at least not initially.) The fact that so little attention has been paid to this issue for this long suggests to me that the Windows user experience is perhaps not highly prioritized by the Atom team, which makes me a bit leery of committing time to learning how to use Atom.

Sorry. This seems like a nice app in general, and I usually prefer to support Open Source when possible (and I do my part by submitting bug reports, etc.), but as I said, true dark mode (not just a dark theme) is a dealbreaker for me, and I'm sure many other people. Having one relatively unimportant part of my screen be bright white while the rest of the screen is dark is visually distracting and makes it difficult for me to "tune out" items that I am not working on at the moment. Also, I get migraines easily, so this is a health issue for me. 

If this feature does get implemented, I might take another look at Atom, but unfortunately, once I have paid for a commercial app like Sublime, etc., it's unlikely that I will use Notepad++ or Atom again, unless I encounter significant problems.

I hope this issue does get fixed soon, though, if not for me, then for the sake of other Windows users of Atom.

**Comment by DanielTamkin** [comment]
I agree @UnchartedSystems, been using Atom for years and love it. But the lack of title-bar support seriously effects the functionality now. Hell, _Slack_ fully supports Electrons new API's for Windows 10 title-bars. Running @sindrets's plugin has always been a delight, but the last update was back in 2019... a solid feature should be standard in the codebase to prevent any hangs/weird bugs caused by plugins recompiling the whole program.

**Comment by sadick254** [comment]
Thanks for reporting this!

This has already been fixed in https://github.com/atom/atom/pull/21234. It is currently scheduled for release when `1.53` reaches Stable.

**Comment by DanielTamkin** [comment]
Wicked! Thanks @TheDrawingCoder-Gamer & @sadick254 😊

So excited to check this out!

**Comment by tonglei100** [comment]
how to set the bar's color dark? or hidden the title bar?

**Comment by binaryfunt** [comment]
This issue has not been addressed. A hidden titlebar is not the same as a custom titlebar.

**Comment by TheDrawingCoder-Gamer** [comment]
Atom is so hackable that allowing the title bar to be hidden allows plugins to add their own title bar, thus making a custom title bar.

**Comment by binaryfunt** [comment]
Yes, sorry, have deleted my previous reply. It seems to work well with https://atom.io/packages/title-bar-replacer
